modules/trionBidAdapter.js

```diff
@@ -1,129 +1,153 @@
-var CONSTANTS = require('src/constants.json');
-var utils = require('src/utils.js');
-var adloader = require('src/adloader.js');
-var bidmanager = require('src/bidmanager.js');
-var bidfactory = require('src/bidfactory.js');
-var Adapter = require('src/adapter.js').default;
-var adaptermanager = require('src/adaptermanager');
-
-const BID_REQUEST_BASE_URL = 'https://in-appadvertising.com/api/bidRequest?';
-const USER_SYNC_URL = 'https://in-appadvertising.com/api/userSync.js';
-
-function TrionAdapter() {
-  var baseAdapter = new Adapter('trion');
-  var userTag = null;
-
-  baseAdapter.callBids = function (params) {
-    var bids = params.bids || [];
-
-    if (!bids.length) {
-      return;
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+const BID_REQUEST_BASE_URL = 'https://in-appadvertising.com/api/bidRequest';
+const USER_SYNC_URL = 'https://in-appadvertising.com/api/userSync.html';
+const BIDDER_CODE = 'trion';
+const BASE_KEY = '_trion_';
+
+export const spec = {
+  code: BIDDER_CODE,
+  isBidRequestValid: function (bid) {
+    return !!(bid && bid.params && bid.params.pubId && bid.params.sectionId);
+  },
+  buildRequests: function (validBidRequests) {
+    var bidRequests = [];
+
+    for (var i = 0; i < validBidRequests.length; i++) {
+      var bid = validBidRequests[i];
+
+      var trionUrlParams = buildTrionUrlParams(bid);
+
+      bidRequests.push({
+        method: 'GET',
+        url: BID_REQUEST_BASE_URL,
+        bidRequest: bid,
+        data: trionUrlParams
+      });
     }
+    return bidRequests;
+  },
 
-    if (!window.TRION_INT) {
-      adloader.loadScript(USER_SYNC_URL, function () {
-        userTag = window.TRION_INT || {};
-        userTag.pubId = utils.getBidIdParameter('pubId', bids[0].params);
-        userTag.sectionId = utils.getBidIdParameter('sectionId', bids[0].params);
-        if (!userTag.to) {
-          getBids(bids);
-        } else {
-          setTimeout(function () {
-            getBids(bids);
-          }, userTag.to);
-        }
-      }, true);
-    } else {
-      userTag = window.TRION_INT;
-      getBids(bids);
-    }
-  };
+  interpretResponse: function (trionResponseObj, request) {
+    var bid = {};
+    var bidResponses = [];
+    var bidRequest = request.bidRequest;
```

Comment:
Where is this coming from?  I think you're doing something similar to what I did in the rubicon adapter to get the original bidRequest, but I attached it to the request;  I don't see you doing the same in this code.

Reply:
Hey @snapwich, the bidrequest is included in the build request function on line 25 of the adapter.  I'm sending it up with the bid request object for each bid.  The request is coming through on all the tests on my end.  Let me know if there is something I'm missing.